public/js/shell.js

```diff
@@ -49,8 +49,8 @@ export default class Shell {
     tab.classList.remove('activetab');
   }
 
-  afterAttach(page) {
-    const options = this._states.get(page);
+  updateState(route) {
```

Comment:
suggestion: s/updateState/onRouteChange/ -> updateState is an implementation detail, onRouteChange makes it more obvious when this should be called.